src/rust/engine/fs/store/src/local.rs

```diff
@@ -308,6 +308,7 @@ impl UnderlyingByteStore for ShardedFSDB {
   }
 
   async fn remove(&self, fingerprint: Fingerprint) -> Result<bool, String> {
+    let _ = self.dest_initializer.lock().remove(&fingerprint);
```

Comment:
To make `dest_initializer` process safe, it looks like you'll want to succeed on `rename` iff the destination already exists. But could be a separate PR.

Reply:
Inside of `write_using`?

Reply:
Yea.

Reply:
`rename` already succeeds by overwriting! Oops. TIL.